src/Dfe.PlanTech.Application/Cookie/Service/CookiesCleaner.cs

```diff
@@ -53,5 +53,6 @@ private static CookieOptions CreateExpiryCookieOptions()
     /// <param name="context"></param>
     /// <returns></returns>
     private IEnumerable<string> GetNonEssentialCookies(HttpContext context)
-      => context.Request.Cookies.Where(c => !opts.EssentialCookies.Any(e => c.Key.StartsWith(e))).Select(cookie => cookie.Key);
+      => context.Request.Cookies.Where(cookie => !Array.Exists(opts.EssentialCookies, essentialCookie => cookie.Key.StartsWith(essentialCookie)))
```

Comment:
Sonar cloud insisted on this instead of `.Any`!? I would have argued for the former 🤔

Reply:
I think it just hates `.Any()`. I would obviously agree (hence the original code) but I'm tired of arguing with a machine :(

Reply:
To add to this, I think from a readability perspective, if we were doing something completely different like 

```csharp
private bool GetNonEssentialCookies(HttpContext context)
  => opts.EssentialCookies.Any(essential => true);
```

then exists would be more performant and just as readable:

```csharp
private bool GetNonEssentialCookies(HttpContext context)
  => Array.Exists(opts.EssentialCookies, cookie => true);
```

Alternatively, if `opts.EssentialCookies` was a _List_ (which has its own `Exists` method), then our actual implementaiton could use it much nicer:

```csharp
private bool GetNonEssentialCookies(HttpContext context)
  => context.Request.Cookies.Where(c => !opts.EssentialCookies.Exists(e => c.Key.StartsWith(e))).Select(cookie => cookie.Key);
```

But SonarCloud doesn't look at the full picture I guess, it just goes "ANY IS NOT AS GOOD AS EXISTS SO PLEASE USE THAT" without looking at the readability of the whole statement.

Reply:
Yeah, 100%, I don't agree with its choice here, but the suggestion isn't that bad I suppose, and I'm in favour of silencing it, so yeah lets just go with it